src/main/java/com/dynatrace/oneagent/sdk/impl/proxy/OneAgentSDKProxy.java

```diff
@@ -268,4 +272,32 @@ public IncomingMessageProcessTracer traceIncomingMessageProcess(MessagingSystemI
 		return new IncomingMessageProcessTracerProxy(apiProxy, agentObject);
 	}
 
+	@Override
+	public DatabaseInfo createDatabaseInfo(String name, String vendor, ChannelType channelType,
+			String channelEndpoint) {
+		return new DatabaseInfoImpl(name, vendor, channelType, channelEndpoint);
+	}
+
+	@Override
+	public DatabaseRequestTracer traceSqlDatabaseRequest(DatabaseInfo databaseInfo, String statement) {
+		if (databaseInfo instanceof DatabaseInfoNoop) {
+			return DatabaseRequestTracerNoop.INSTANCE;
+		} else if (!(databaseInfo instanceof DatabaseInfoImpl)) {
+			if (OneAgentSDKFactoryImpl.debugOneAgentSdkStub) {
+				OneAgentSDKFactoryImpl.logDebug("- invalid DatabaseInfo object provided: "
+						+ (databaseInfo == null ? "null" : databaseInfo.getClass().getName()));
+			}
+			return DatabaseRequestTracerNoop.INSTANCE;
+		}
+
+		Object agentObject = apiProxy.oneAgentSDK_traceSQLDatabaseRequest(agentSdkImpl, (DatabaseInfoImpl) databaseInfo, statement);
+		if (agentObject == null) {
+			if (OneAgentSDKFactoryImpl.debugOneAgentSdkStub) {
+				OneAgentSDKFactoryImpl.logDebug("- OneAgent failed to provide provide object");
```

Comment:
typo "provide provide". Also, maybe "failed to create database request trace object" would be more informative. And why the leading "-"?

Reply:
thank - will be included in next release